error.go

```diff
@@ -0,0 +1,27 @@
+package microstorage
+
+import (
+	"github.com/juju/errgo"
+)
+
+var InvalidConfigError = errgo.New("invalid config")
+
+// IsInvalidConfig represents the error matcher for public use. Code using the
+// storage internally should use this public key matcher to verify if some
+// storage error is of type "key not found", instead of using a specific error
+// matching of some specific storage implementation. This public error matcher
+// groups all necessary error matchers of more specific storage
+// implementations.
+func IsInvalidConfig(err error) bool {
+	return errgo.Cause(err) == InvalidConfigError
+}
+
+var NotFoundError = errgo.New("not found")
+
+// IsNotFound represents the error matcher for public use. Code using the
+// storage internally should use this public key matcher to verify if some
+// storage error is of type "key not found", instead of using a specific error
+// matching of some specific storage implementation.
```

Comment:
The above is actually a comment I wrote somewhere. Its intention was to make sure users do not use special implementation matchers of sub packages. Here we have all implementations in the same package, so the comment does not hold true anymore, since the implementation of the matcher here did change as well. The comment suggestion you just posted is better. Lets just get rid of the "special implementation" thing because this is not the case here.